activerecord/lib/active_record/core.rb

```diff
@@ -534,17 +533,18 @@ def pretty_print(pp)
       return super if custom_inspect_method_defined?
       pp.object_address_group(self) do
         if defined?(@attributes) && @attributes
-          column_names = self.class.column_names.select { |name| has_attribute?(name) || new_record? }
-          pp.seplist(column_names, proc { pp.text "," }) do |column_name|
+          attr_names = self.class.attribute_names.select { |name| has_attribute?(name) }
+          pp.seplist(attr_names, proc { pp.text "," }) do |attr_name|
             pp.breakable " "
             pp.group(1) do
-              pp.text column_name
+              pp.text attr_name
               pp.text ":"
               pp.breakable
-              if filter_attribute?(column_name)
-                pp.text ActiveRecord::Core::FILTERED
+              value = inspection_filter.filter_param(attr_name, read_attribute(attr_name))
+              if value == ActiveSupport::ParameterFilter::FILTERED
```

Comment:
If `read_attribute(attr_name)` returns `"[FILTERED]"`, this condition would be true too even if the attribute is not filtered.